sensor/kubernetes/listener/resources/networkpolicy_test.go

```diff
@@ -0,0 +1,445 @@
+				Name:      "deploy-4",
+				Id:        "4",
+				Namespace: "default",
+				PodLabels: map[string]string{
+					"app": "sensor-2",
+				},
+			},
+		},
+	}
+
+	for _, d := range deployments {
+		ds.addOrUpdateDeployment(d)
+	}
+
+	cases := []struct {
+		name                string
+		netpol              *storage.NetworkPolicy
+		sel                 []map[string]string
+		expectedDeployments []*deploymentWrap
+	}{
+		{
+			name: "New NetworkPolicy",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			sel: []map[string]string{
+				{
+					"app":  "sensor",
+					"role": "backend",
+				},
+			},
+			expectedDeployments: []*deploymentWrap{
+				{
+					Deployment: &storage.Deployment{
+						Id:        "1",
+						Namespace: "default",
+					},
+				},
+			},
+		},
+		{
+			name: "Empty selector",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			sel: []map[string]string{},
+			expectedDeployments: []*deploymentWrap{
+				{
+					Deployment: &storage.Deployment{
+						Id:        "1",
+						Namespace: "default",
+					},
+				},
+				{
+					Deployment: &storage.Deployment{
+						Id:        "2",
+						Namespace: "default",
+					},
+				},
+				{
+					Deployment: &storage.Deployment{
+						Id:        "4",
+						Namespace: "default",
+					},
+				},
+			},
+		},
+		{
+			name: "Selector with no deployments",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			sel: []map[string]string{
+				{
+					"app": "central",
+				},
+			},
+			expectedDeployments: []*deploymentWrap{},
+		},
+		{
+			name: "Namespace with no deployments",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "random_namespace",
+			},
+			sel: []map[string]string{
+				{
+					"app": "sensor",
+				},
+			},
+			expectedDeployments: []*deploymentWrap{},
+		},
+		{
+			name: "Namespace with no deployments, no selector",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "random_namespace",
+			},
+			sel:                 []map[string]string{},
+			expectedDeployments: []*deploymentWrap{},
+		},
+		{
+			name: "Disjunction selector",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			sel: []map[string]string{
+				{
+					"app":  "sensor",
+					"role": "backend",
+				},
+				{
+					"app": "sensor-2",
+				},
+			},
+			expectedDeployments: []*deploymentWrap{
+				{
+					Deployment: &storage.Deployment{
+						Id:        "1",
+						Namespace: "default",
+					},
+				},
+				{
+					Deployment: &storage.Deployment{
+						Id:        "4",
+						Namespace: "default",
+					},
+				},
+			},
+		},
+	}
+	for _, c := range cases {
+		deps := map[string]*deploymentWrap{}
+		processDeploymentMock := det.EXPECT().ProcessDeployment(gomock.Any(), central.ResourceAction_UPDATE_RESOURCE).DoAndReturn(func(d *storage.Deployment, _ central.ResourceAction) {
```

Comment:
AFAIK the expectation needs to be defined with a matcher other than the actual value. 
```suggestion
		processDeploymentMock := det.EXPECT().ProcessDeployment(gomock.Any(), gomock.Eq(central.ResourceAction_UPDATE_RESOURCE)).DoAndReturn(func(d *storage.Deployment, _ central.ResourceAction) {
```

Reply:
Oops... I've fixed it 😅